app/renderer/actions/Inspector.js

```diff
@@ -137,26 +137,24 @@ export function selectElement (path) {
   return async (dispatch, getState) => {
     // Set the selected element in the source tree
     dispatch({type: SELECT_ELEMENT, path});
-    const state = getState().inspector;
-    const {attributes: selectedElementAttributes, xpath: selectedElementXPath} = state.selectedElement;
-    const {sourceXML} = state;
+    const { selectedElement, sourceXML, expandedPaths } = getState().inspector;
 
     // Expand all of this element's ancestors so that it's visible in the source tree
-    let {expandedPaths} = getState().inspector;
+    const copiedExpandedPaths = expandedPaths.slice();
     let pathArr = path.split('.').slice(0, path.length - 1);
     while (pathArr.length > 1) {
       pathArr.splice(pathArr.length - 1);
       let path = pathArr.join('.');
-      if (expandedPaths.indexOf(path) < 0) {
-        expandedPaths.push(path);
+      if (copiedExpandedPaths.indexOf(path) < 0) {
```

Comment:
could also be (!copiedExpandedPaths.includes(path))

Reply:
I didn't really check this condition, but I agree that using not include makes it a lot clearer.